msdk-featdet/msdk-featdet-ADAP-3D/src/main/java/io/github/msdk/featdet/ADAP3D/common/algorithms/CurveTool.java

```diff
@@ -130,24 +130,46 @@ public static double similarityValue(double[] referenceEIC, double[] gaussianVal
    * @return area a {@link java.lang.Double} object. This is area of normalize intensity points.
    *         </p>
    */
-  public static double normalize(MultiKeyMap<Integer, Triplet> slice, int leftBound, int rightBound,
-      int roundedMz, double[] referenceEIC) {
-
+  public static double normalize(List<Triplet> slice, int leftBound, int rightBound, int roundedMz,
+      double[] referenceEIC) {
+
+    Comparator<Triplet> compare = new Comparator<Triplet>() {
+
+      @Override
+      public int compare(Triplet o1, Triplet o2) {
+        int scan1 = o1.scanNumber;
+        int scan2 = o2.scanNumber;
+        int scanCompare = Integer.compare(scan1, scan2);
+
+        if (scanCompare != 0) {
+          return scanCompare;
+        } else {
+          int mz1 = o1.mz;
+          int mz2 = o2.mz;
+          return Integer.compare(mz1, mz2);
+        }
+      }
+    };
```

Comment:
Thanks for review. on many places I'm sorting this list on different parameters.